website/docs/quickstart.md

```diff
@@ -0,0 +1,211 @@
+
+Now, without further ado, let's get started.
+
+To do our build, we need a few things:
+
+1. A list of sources to pull from
+2. A build script to build the sources
+3. A list of artifacts to include in the package
+
+In this example, we'll build the `go-md2man` package and container from the [`go-md2man`](https://github.com/cpuguy83/go-md2man) repo using `v2.0.3` tag in the repo.
+
+First, let's start with the constructing a [Dalec spec](spec.md) file.
+
+We define the metadata for the package in the spec. This includes the name, packager, vendor, license, website, and description of the package. You may notice that many of these fields appear in package manager metadata for rpm and deb packages. This is because, for certain targets, Dalec will generate package files for these package managers and will utilize this information for that purpose.
+
+```yaml
+# syntax=ghcr.io/azure/dalec/frontend:latest
+name: go-md2man
+version: 2.0.3
+revision: "1"
+license: MIT
+description: A tool to convert markdown into man pages (roff).
+packager: Dalec Example
+vendor: Dalec Example
+website: https://github.com/cpuguy83/go-md2man
+```
+
+:::tip
+In metadata section, `packager`, `vendor` and `website` are optional fields.
+:::
+
+In the next section of the spec, we define the [sources](sources.md) that we will be pulling from. In this case, we are pulling from a git repository.
+
+One thing to note: in many build systems you will not have access to the Internet while building the package, and indeed that is the case with the `mariner2` target. As such, this build will fail because `go build` will try to download the go modules. For this reason, we added a `generate` section to the source to run `go mod download` in a docker image with the `src` source mounted and then extract the go modules from the resulting filesystem.
+
+```yaml
+sources:
+  src:
+    git:
+      url: https://github.com/cpuguy83/go-md2man.git
+      commit: "v2.0.3"
+    generate:
+      - gomod: {}
+```
+
+In the next section, we define the dependencies that are needed to build the package. In this case, we need the `golang` dependency at the build time, and `openssl-libs` at runtime. Build dependencies are dependencies that are needed to build the package, while runtime dependencies are dependencies that are needed to run the package, i.e., they will be installed alongside the package when it is installed on a system. Runtime dependencies are not required for this specific example, but they are included for illustrative purposes.
+
+```yaml
+dependencies:
+  build:
+    golang:
+  runtime:
+    # as stated above, included for illustrative purposes
+    openssl-libs:
+```
+
+Now, let's define the build steps. In this case, we are building the `go-md2man` binary.
+
+```yaml
+build:
+  # the env section allows us to define environment variables 
+  # that will be set during the build process
+  env:
+    CGO_ENABLED: "0"
+  steps:
+    - command: |
+        cd src
```

Comment:
Maybe we can add a note as to where `src` comes from.

Reply:
+1 this has been confusing for me in the past for sure